library/src/main/scala/giter8/launcher/Launcher.scala

```diff
@@ -0,0 +1,132 @@
+package giter8.launcher
+
+import java.io.{File, FileInputStream}
+import java.util.Properties
+
+import giter8.{TempDir, _}
+import org.apache.commons.io.FileUtils
+
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+object Launcher extends TempDir {
+
+  import giter8.G8.RichFile
+  import giter8.OptTry
+
+  private val git = new Git(new JGitInteractor())
+
+  /**
+    * Use JAVA_HOME as a default location for finding a java executable. If
+    * it isn't set fall back to the `java.home` of the current JVM
+    */
+  private lazy val javaHome: String =
+    sys.env.getOrElse("JAVA_HOME", System.getProperty("java.home"))
+
+  /**
+    * Resolves the version of giter8 to be launched.
+    *
+    * If `g8Version` is defined, that will be used. Otherwise, we clone
+    * the template repo and look for a `giter8.version` property in `project/build.properties`
+    *
+    * Returns `None` if we can't find a version from either of these
+    *
+    * @param g8Version
+    * @param template
+    * @return
+    */
+  private def version(g8Version: Option[String], template: String): Try[Option[VersionNumber]] =
+    Try(g8Version.map(VersionNumber.apply)) or git.withRepo(template) {
+      base =>
+        Try {
+          val properties: Properties = {
+            val p = new Properties()
+            p.load(new FileInputStream(base / "project" / "build.properties"))
+            p
+          }
+          val result: Option[String] = Option(properties.getProperty("giter8.version"))
+          result
+        }.flatMap {
+          string =>
+            Try(string.map(VersionNumber.apply))
+        }
+    }.flatten
+
+  /**
+    * Retrieves the launch config of a specific tagged version of Giter8,
+    * if `version` is undefined, we attempt to retrieve the latest version
+    *
+    * NOTE: It seems that the version number in some of the launchconfigs
+    * does not match the release version. (I think that this is due to 0.8.0 being
+    * reverted, so it shouldn't be a problem)
+    *
+    * e.g. https://github.com/foundweekends/giter8/blob/v0.9.0/src/main/conscript/g8/launchconfig#L2
+    *
+    * @param version
+    * @return
+    */
+  private def launchConfig(version: Option[VersionNumber]): Try[String] =
```

Comment:
Could this mean that we need to have a GitHub-public tag for all of the Giter8 versions?
I think it's ok to fallback to the whatever the current one in-memory when it can't find it, for example to support local testing.

Reply:
I'm fine falling back to the current version but I think that might be a problem from `sbt new` as technically we're launching it from outside of Giter8.